components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -569,6 +574,43 @@ void EthereumProviderImpl::RecoverAddress(const std::string& message,
                           false);
 }
 
+void EthereumProviderImpl::EthSubscribe(const std::string& event_type,
+                                        RequestCallback callback,
+                                        base::Value id) {
+  if (event_type != kEthSubscribeNewHeads) {
+    base::Value formed_response = GetProviderErrorDictionary(
+        mojom::ProviderError::kInternalError,
+        l10n_util::GetStringUTF8(IDS_WALLET_UNSUPPORTED_SUBSCRIPTION_TYPE));
+    bool reject = true;
+    std::move(callback).Run(std::move(id), std::move(formed_response), reject,
+                            "", false);
+    return;
+  }
+  std::vector<uint8_t> bytes(16);
+  crypto::RandBytes(&bytes.front(), bytes.size());
+  std::string hex_bytes = ToHex(bytes);
+  eth_subscriptions_.push_back(hex_bytes);
+  if (eth_subscriptions_.size() == 1)
+    eth_block_tracker_.Start(base::Seconds(10));
```

Comment:
any reason that this is shorter than `kBlockTrackerDefaultTimeInSeconds` (20)?

Reply:
Fixed / rebased in e1d053ba7448d9544d3ccbf92b1ebf2ce299df38